src/codeu/chat/client/simplegui/ChatGuiFX.java

```diff
@@ -199,8 +191,30 @@ public void start(Stage primaryStage) throws Exception {
         thestage.show();
     }
 
-    private void buttonClicked(ActionEvent e)
-    {
+    private void buttonClicked(ActionEvent e) {
         thestage.setScene(mainScene);           // TODO: Call a controller function here instead
     }
+
+    private void addConversation(ActionEvent e) {
+        // popup for the user to add a conversation
+        TextInputDialog dialog = new TextInputDialog("Name your conversation!");
+        dialog.setTitle(" "); // make the title blank
+        dialog.setHeaderText("Create your conversation");
+
+        // get input and add the name of the convo
+        String name = dialog.showAndWait().get();
+
+        // TODO: check for duplicate conversations & handle them
+
+        if (name != null && name.length() > 0) {
+            // TODO: users need to be added to the server first? So I can add the conversation to that user???
+            //clientContext.conversation.startConversation(name, clientContext.user.getCurrent().id);
+            convoList.addAll(name);
+        }
+    }
+
+    private void selectConversation(MouseEvent e) {
```

Comment:
At some point we'd like to add users to the conversation, and only they would have permission to participate in it. The basis of this functionality has been implemented in the database, but we need to add code on the front-end and back-end to actually make this happen.